quick_tour/the_architecture.rst

```diff
@@ -1,4 +1,139 @@
 The Architecture
 ================
 
-TODO
+You are my hero! Who would have thought that you would still be here after the first
+two parts? Your efforts will be well-rewarded soon. The first two parts didn't look
+too deeply at the architecture of the framework. Because it makes Symfony stand apart
+from the framework crowd, let's dive into the architecture now.
+
+Add Logging
+-----------
+
+A new Symfony app is micro: it's basically just a routing & controller system. But
+thanks to Flex, installing more features is simple.
+
+Want a logging system? No problem:
+
+.. code-block:: terminal
+
+    $ composer require logger
+
+This installs and configures (via a recipe) the powerful `Monolog`_ library. To
+use the logger in a controller, add a new argument type-hinted with ``LoggerInterface``:
+
+    use Psr\Log\LoggerInterface;
+    // ...
+
+    public function index($name, LoggerInterface $logger)
+    {
+        $logger->info("Saying hello to $name!");
+
+        // ...
+    }
+
+That's it! The new log message be written to ``var/log/dev.log``. Of course, this
+can be configured by updating one of the config files added by the recipe.
+
+Services & Autowiring
+---------------------
+
+But wait! Something *very* cool just happened. Symfony read the ``LoggerInterface``
+type-hint and automatically figured out that it should pass us the Logger object!
+This is called *autowiring*.
+
+Every bit of work that's done in a Symfony app is done by an *object*: the Logger
+object logs things and the Twig object renders templates. These objects are called
+*services* and they are the *tools* that make you dangerous.
```

Comment:
I'm not sure this *"dangerous"* word is going to be understood well by people with basic English level, which are most of our readers. Why if we say something easier to understand, like: *"...the *tools* that give you super powers!"*